Add more temporary directory definitions.

laur89 opened the pull request:
As many environments limit the writability of /tmp to regular users, then script
could potentially benefit from writing into user homedir in case there's no write
access to /tmp.

fsquillace commented on lib/ssh_pearl.sh:
```diff
@@ -34,16 +34,19 @@ function ssh_pearl(){
+    local remoteTmpDirs=( "/tmp" "\$HOME" )
 
     CMD="
+        for tmpDir in ${remoteTmpDirs[*]}; do [[ -w \"\$tmpDir\" ]] && { foundTmpDir=\"\$tmpDir\"; break; } done
```
Since the list of directory does not change dynamically, I would place the list explicitly here:

```
for tmpDir in /tmp \"\$HOME\"; do ..
```

laur89 replied:
Thought for the remoteTmpDirs array to act as a sort of in-code configuration, but indeed, the array isn't used dynamically.

fsquillace commented on lib/ssh_pearl.sh:
```diff
@@ -34,16 +34,19 @@ function ssh_pearl(){
 
     CMD="
+        for tmpDir in ${remoteTmpDirs[*]}; do [[ -w \"\$tmpDir\" ]] && { foundTmpDir=\"\$tmpDir\"; break; } done
+        [[ -z \"\$foundTmpDir\" ]] && { echo -e \"couldn't find writable tempdirs on the server. Aborting.\"; exit 1; };
```
Similarly to the other comments I would redirect it to stderr:

```
[[ -z \"\$foundTmpDir\" ]] && { echo >&2 \"Could not find writable tempdirs on the server. Aborting.\"; exit 1; };
```

fsquillace commented:
Awesome change :)

Could you make the suggestion I reported so that it makes the code a bit cleaner. Also, if you can squash the two commits to be just one, it would make the history a bit readable.

```
 git rebase --interactive origin/master
```

Thanks again!
Filippo

Commit:
Add more temporary directory definitions.
As many environments limit the writability of /tmp to regular users, then script
could potentially benefit from writing into user homedir in case there's no write
access to /tmp.

+ Additionally:
    - make syntax more 'bash-like', ie change the test expressions common to sh to
      the bash preferred [[  ]];
    - improve safety by double-quoting all the vars;
    - add more triggering signals for the trap;

laur89 commented:
5b7d3ca should be good for merge now.

fsquillace commented:
Merged. Really nice code change :)

Could you star this project if you like it? I am trying to get this project more popular.

Cheers,
Filippo

laur89 commented:
Yeah sure, this **should** be more popular.
